vm/resolutions.go

```diff
@@ -139,8 +139,12 @@ func (vm *VM) Verified(ctx context.Context, b *StatefulBlock) {
 	vm.verifiedBlocks[b.ID()] = b
 	vm.verifiedL.Unlock()
 	vm.parsedBlocks.Evict(b.ID())
-	vm.mempool.Remove(ctx, b.StatelessBlock.Txs)
-	vm.gossiper.BlockVerified(b.Tmstmp)
+
+	for _, sub := range vm.verifiedSubscriptions {
+		if err := sub.Accept(b.executedBlock); err != nil {
+			vm.Fatal("subscription failed to process verified block", zap.Error(err))
+		}
+	}
```

Comment:
We do this looping in a bunch of places, should we define a helper that looks like:
```golang
func NotifyAll[T any](e T, subs ...Subscription[T]) error {
	var errs []error
	for _, sub := range subs {
		if err := sub.Accept(e); err != nil {
			errs = append(errs, err)
		}
	}
	return errors.Join(errs)
}
```

Reply:
Updated